velox/functions/prestosql/tests/BigintEnumCastTest.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/CastBaseTest.h"
+#include "velox/functions/prestosql/types/BigintEnumRegistration.h"
+#include "velox/functions/prestosql/types/BigintEnumType.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec;
+
+namespace {
+
+class BigintEnumCastTest : public functions::test::CastBaseTest {
+ protected:
+  BigintEnumCastTest() {
+    registerBigintEnumType();
+  }
+};
+
+TEST_F(BigintEnumCastTest, toBigintEnum) {
+  auto enumName = "test.enum.mood";
+  std::unordered_map<std::string, int64_t> enumMap = {
+      {"CURIOUS", -2}, {"HAPPY", 0}};
+  LongEnumParameter longEnumParameter(enumName, enumMap);
+  const std::vector<TypeParameter>& typeParameters = {
+      TypeParameter(longEnumParameter)};
+
+  // Cast base type to enum type
+  testCast<int64_t, int64_t>(
+      BIGINT(),
+      BIGINT_ENUM(typeParameters),
```

Comment:
Let's add a variable for this type to avoid copy-paste.

```
auto moodEnum = BIGINT_ENUM(LongEnumParameter{"mood", {{"HAPPY", 1}, {"SAD", 2}}})
```